src/ad.jl

```diff
@@ -0,0 +1,9 @@
+function _abs_deriv(x::Interval{T}) where T 
```

Comment:
I believe to support decorated interval, we just need the definition

```julia
_abs_deriv(x::DecoratedInterval) = DecoratedInterval(_abs_deriv(interval(x)), decoration(x))
```
since this operation can not change the decoration (i.e. the validity) of the operation.